app/src/main/java/com/wednesday/banklocator/fragment/BankSearchFragment.kt

```diff
@@ -1,7 +1,64 @@
 package com.wednesday.banklocator.fragment
 
+import android.os.Bundle
+import android.util.Log
+import android.view.View
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.LinearLayoutManager
 import com.wednesday.banklocator.R
+import com.wednesday.banklocator.activity.MainActivity
+import com.wednesday.banklocator.adapter.IfscAdapter
+import com.wednesday.banklocator.util.Resource
+import com.wednesday.banklocator.viewmodel.BankViewModel
+import kotlinx.android.synthetic.main.fragment_search_bank.*
+
+
+class BankSearchFragment : Fragment(R.layout.fragment_search_bank) {
+    lateinit var viewModel: BankViewModel
+    lateinit var newsIfscAdapter: IfscAdapter
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        //setupRecyclerView()
+        viewModel = (activity as MainActivity).viewModel
+        searchButton.setOnClickListener {
+            if (etSearch2.toString().isNotEmpty()) {
+                viewModel.getIfscCode(etSearch2.text.toString())
+            }
+        }
+        viewModel.ifscCodes.observe(viewLifecycleOwner, { response ->
+            when (response) {
+                is Resource.Success -> {
+                    hideProgressBar()
+                    response.data?.let { newsResponse ->
+                        newsIfscAdapter = IfscAdapter(newsResponse)
```

Comment:
Everytime you get a response you should not create an adapter. You should instead pass the new data to the adapter and ask it to reset it's datasource using `notifyDataSetChange` this will avoid the need to keep allocating new objects.

Reply:
I'll implement "notifyDataSetChange" now.